fcm/fcm.go

```diff
@@ -99,22 +99,33 @@ func (f *FCM) ConnectToFcm(fcmNotifications chan<- notification.PrinterNotificat
 		go func() {
 			for {
 				raw_input, err1 := reader.ReadBytes('\n')
-				inputA := strings.SplitN(string(raw_input), "\\n", 2)
-				notification_size := inputA[0]
+				input_chunks := strings.SplitN(string(raw_input), "\n", 2)
+				notification_size := strings.TrimSpace(input_chunks[0])
 				notification_data1 := ""
-				if len(inputA) > 1 {
-					notification_data1 = inputA[1]
+				if len(input_chunks) > 1 {
+					notification_data1 = strings.TrimSpace(input_chunks[1])
 				}
-				size, _ := strconv.Atoi(strings.TrimSpace(notification_size))
+				size, _ := strconv.Atoi(notification_size)
 				buffer_size := size - len(notification_data1)
 
-				// part of notification
-				notification_data2_buffer := make([]byte, buffer_size)
-				_, err2 := reader.Read(notification_data2_buffer)
-				if err1 == io.EOF  || err2 == io.EOF || (err2 == nil && err1 == nil) {
-					notificationS := string(notification_data2_buffer) + notification_data1
-					if len(notificationS) > 0  {
-						printerId := GetPrinterID(notificationS)
+				notification_data2 := ""
+				var err2 error
+				for 0 != buffer_size {
+					// part of notification
+					notification_data2_buffer := make([]byte, buffer_size)
+					n, err2 := reader.Read(notification_data2_buffer)
+					notification_data2 += string(notification_data2_buffer)
```

Comment:
I think conversion to string from a buffer that contains \0 bytes will embed them in the string, right? So you want to embed only the first n bytes of the buffer into the string. I just sent you on email a code snippet to look at. I think ideally you would Read() into the same buffer until it's over or you get an error, after that you can convert to string if needed.

Reply:
Appending to the buffer instead of string concatenation.
Fixed.